partner_preview_scripts/ValidationLab.py

```diff
@@ -3,432 +3,472 @@
+            "name": "KRAS-",
+        },
+        "TP53_status": {
+            "description": "TP53 mutation status: ",
+            "name": "TP53-",
+        },
+        "APC_status": {
+            "description": "APC mutation status: ",
+            "name": "APC-",
+        },
+        "BRAF_status": {"description": "BRAF mutation status: ", "name": "BRAF-"},
+    }
+
+    @staticmethod
+    def _get_biomarker_wrapper(biomarker_map: dict) -> Callable:
+        @f.udf(
+            t.StructType(
+                [
+                    t.StructField("name", t.StringType(), False),
+                    t.StructField("description", t.StringType(), False),
+                ]
             )
-            # The gene names are manually typed, there are lower and upper case names:
-            .withColumn("gene", f.upper(f.col("gene")))
         )
+        def wrapped_function(column_name: Column, biomarker: Column) -> dict | None:
+            """This function returns a struct with the biomarker name and description."""
+            # If the biomarker is not applied on the given cell type the value is zero:
+            if biomarker == "0":
+                return None
+
+            # If the biomarker value has a direct mapping:
+            if "direct_mapping" in biomarker_map[column_name]:
+                try:
+                    return biomarker_map[column_name]["direct_mapping"][biomarker]
+                except KeyError:
+                    logging.warning(
+                        f"Could not find direct mapping for {column_name}:{biomarker}"
+                    )
+                    return None
+
+            # If the value needs to be parsed:
+            if biomarker == "wt":
+                return {
+                    "name": biomarker_map[column_name]["name"] + biomarker,
+                    "description": biomarker_map[column_name]["description"]
+                    + "wild type",
+                }
+            elif biomarker == "mut":
+                return {
+                    "name": biomarker_map[column_name]["name"] + biomarker,
+                    "description": biomarker_map[column_name]["description"] + "mutant",
+                }
+            else:
+                logging.warning(
+                    f"Could not find direct mapping for {column_name}: {biomarker}"
+                )
+                return None
+
+        return wrapped_function
+
+    @classmethod
+    def get_biomarkers(
+        cls: type[BiomarkerParser], biomarker_df: DataFrame
+    ) -> DataFrame:
+        biomarkers_in_data = [
+            biomarker
+            for biomarker in cls.BIOMARKERMAPS.keys()
+            if biomarker in biomarker_df.columns
+        ]
 
-        # The first column is the gene name, the rest are the hypotheses:
-        hypothesis_columns = hypothesis_df.columns[1:]
+        # Applying the full map on the dataframe one-by-one:
+        processed_biomarkers = reduce(
```

Comment:
map+reduce are too complex for my brain, but if the intention is to create a new column per biomarker, could you just iterate over the list of those biomarkers present in the data? Sth like
```
  for biomarker in biomarkers_to_process:
    processed_biomarker = cls._get_biomarker_wrapper(biomarker_maps)(
        f.lit(biomarker), f.col(biomarker)
    )
    biomarker_df = biomarker_df.withColumn(biomarker, processed_biomarker)
```

Reply:
fine.